components/campaign/donation/wallet/process.tsx

```diff
@@ -61,21 +61,14 @@ export function CampaignDonationWalletProcess({
     onStateChanged: setState,
   });
   const isValidEmail = (email: string) => {
+    if (email.trim() === '') {
+      return true;
+    }
     return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
   };
 
   const onDonateStart = useCallback(async () => {
-    // Validate email first
-    if (!email.trim()) {
-      toast({
-        title: 'Email required',
-        description: 'Please enter your email address to continue.',
-        variant: 'destructive',
-      });
-      return;
-    }
-
-    if (!isValidEmail(email)) {
+    if (profile?.email && profile.email.trim() !== '' && !isValidEmail(email)) {
```

Comment:
I believe this is a typo and you meant to check the email not the profile email.
```suggestion
    if (email && !isValidEmail(email)) {
```

Note: in line 87 we are updating the `profile.email` if the user doesn't have it set already without checking the value of `email`

- if there is no `profile.email` and `email` is empty, we don't need to query the db.